### PR TITLE
STITCH-1999 Package up sync package with new mobile dependencies

### PR DESCRIPTION
**Pull request by jsflax**
Originally, because there was no mongo driver in this ecosystem, we had to source our own dependencies and create our own `.frameworks`. Now that these frameworks are released as pods, we can remove all the cruft that set up our dev environment.

- Remove all setup scripts
- Add core swift .xcodeproj's
- Use versioned MongoSwift and update source appropriately (e.g., `binary` => `binaryDeprecated`)
- Update podspecs for next release (4.0.7)
- Use MongoMobile for StitchCoreLocalMongoDBService
- Update source for iOS 12.0
- Add cocoapods support for developer environment
- Add cocoapods support for evg
- Update READMEs for new setup.

The only setup a developer will have to do going forward is running `pod install`. Will save us a lot of trouble.

### CODE REVIEW

File: Core/StitchCoreSDK/Sources/StitchCoreSDK/Auth/Internal/CoreStitchAuth+StitchAuthRequestClient.swift [hunk 1]
```diff
@@ -27,13 +27,12 @@ extension CoreStitchAuth {
...
-                        withServiceErrorCode: .unknown
-                    )
+                        withServiceErrorCode: .unknown)
             }
-            
+
             do {
-                // TODO: until Swift Driver decides on what to do
-                return try BsonDecoder().decode(T.self, from: responseString)
+                // TODO: until Swifdriver decides on what to do
```
Comment (adamchel) [inline Core/StitchCoreSDK/Sources/StitchCoreSDK/Auth/Internal/CoreStitchAuth+StitchAuthRequestClient.swift]: [nit] revert, or remove TODO

File: Darwin/Services/StitchLocalMongoDBService/StitchLocalMongoDBService/LocalMongoClient.swift [hunk 2]
```diff
@@ -61,21 +61,21 @@ private final class MobileMongoDBClientFactory: CoreLocalMongoDBService, Throwin
         NotificationCenter.default.addObserver(
             self,
             selector: #selector(batteryLevelDidChange(_:)),
-            name: .UIDeviceBatteryLevelDidChange,
+            name: UIDevice.batteryLevelDidChangeNotification,
```
Comment (adamchel) [inline Darwin/Services/StitchLocalMongoDBService/StitchLocalMongoDBService/LocalMongoClient.swift]: [q] does this affect backwards compatibility?

File: contrib/README.md [hunk 3]
```diff
@@ -4,52 +4,20 @@
...
+### Working with the existing repository
+
+1. Run `pod install` at the top level directory.
+2. If this fails due to a pod not being found, you may have to run `pod update` first.
+
 ### Adding a new module
 
 To add a new module the Stitch workspace, use this procedure.
 
-1. `mkdir <module name>`
```
Comment (adamchel) [inline contrib/README.md]: [nice] I'm so happy we don't need this anymore

File: Podfile [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+    end
+    
+    target :StitchCoreTestUtils do
+        project 'Core/StitchCoreTestUtils/StitchCoreTestUtils.xcodeproj'
+    
+        inherit! :search_paths
+    end    
+end
+
+post_install do |installer|
```
Comment (adamchel) [inline Podfile]: [nit q] do we still need this?

File: Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/RemoteUpdateResult.swift [hunk 5]
```diff
@@ -10,15 +10,15 @@ public struct RemoteUpdateResult: Decodable {
...
     
-    internal init(matchedCount: Int, modifiedCount: Int, upsertedId: BsonValue?) {
+    internal init(matchedCount: Int, modifiedCount: Int, upsertedId: BSONValue?) {
         self.matchedCount = matchedCount
         self.modifiedCount = modifiedCount
         self.upsertedId = upsertedId
     }
     
-    // Workaround until SWIFT-104 is merged, which will make BsonValue `Decodable`
+    // Workaround until SWIFT-104 is merged, which will make BSONValue `Decodable`
```
Comment (edaniels) [inline Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/RemoteUpdateResult.swift]: [q] That ticket is resolved, can we change anything here?

File: Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/RemoteInsertOneResult.swift [hunk 6]
```diff
@@ -5,13 +5,13 @@ import MongoSwift
...
-    public let insertedId: BsonValue
+    public let insertedId: BSONValue
     
-    internal init(insertedId: BsonValue) {
+    internal init(insertedId: BSONValue) {
         self.insertedId = insertedId
     }
     
-    // Workaround until SWIFT-104 is merged, which will make BsonValue `Decodable`
+    // Workaround until SWIFT-104 is merged, which will make BSONValue `Decodable`
```
Comment (edaniels) [inline Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/RemoteInsertOneResult.swift]: ditto

File: Darwin/Services/StitchLocalMongoDBService/StitchLocalMongoDBService/LocalMongoClient.swift (hunk 2 above)
Comment (jsflax) [reply Darwin/Services/StitchLocalMongoDBService/StitchLocalMongoDBService/LocalMongoClient.swift]: Not for users, but it does for developers.

File: Podfile (hunk 4 above)
Comment (jsflax) [reply Podfile]: Looks like they haven't 😞 . Going to uncomment this block out for now.

**Pull request merged**